### ISSUE TITLE
[BUG] Indentified technologies are not correct

### ISSUE THREAD

**Issue by rromannissen** [konveyor/analyzer-lsp#404]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Konveyor version

0.3.0-beta.2

### Priority

Blocker

### Current Behavior

Technology tags identified for the Tackle Test application don't match the ones identified by Windup and add multiple technologies that are not being used by the application:

![Screenshot from 2023-11-03 17-15-36](https://github.com/konveyor/analyzer-lsp/assets/15966456/9a285180-b9cf-4845-a27b-71eb0af1061e)

![Screenshot from 2023-11-03 17-15-39](https://github.com/konveyor/analyzer-lsp/assets/15966456/9f6448ff-dc28-4827-9fdb-0896d73e5dd2)

### Expected Behavior

The right list of technologies gets identified, as in previous versions of Konveyor:

![Screenshot from 2023-11-03 17-20-43](https://github.com/konveyor/analyzer-lsp/assets/15966456/998a9a30-2cee-4d09-aa8c-878c8a1fc58d)

### How Reproducible

Always (Default)

### Steps To Reproduce

1. Analyze tackle-test app with the configuration available in the Environment section.
2. Go to the *Application Inventory* and click on the tackle-testapp application
3. Click on the *Tags* tab from the *Application Profile* side drawer.

### Environment

```markdown
- Fedora 36
- minikube v1.31.2 (--driver=kvm2 --memory=10g --cpus=3)
- Konveyor 0.3.0-beta2 (also reproduced in previous beta 1 versions)
- [tackle-testapp](https://github.com/konveyor/tackle-testapp) application
- Analysis configuration:
  - Source + Dependencies
  - cloud-readiness, openjdk, linux
  - Application and internal dependencies only
  - Custom rule manually uploaded
  - No additional advanced options
```

### Anything else?

Execution report:

```
id: 1
createTime: 2023-11-03T11:07:35.978934324Z
name: Customers.1.windup
addon: analyzer
data:
    mode:
        artifact: ""
        binary: false
        diva: false
        withDeps: true
    rules:
        labels:
            excluded: []
            included:
                - konveyor.io/target=cloud-readiness
                - konveyor.io/target=openjdk
                - konveyor.io/target=linux
        path: /rules
        tags:
            excluded: []
    scope:
        packages:
            excluded: []
            included: []
        withKnownLibs: false
    sources: []
    tagger:
        enabled: true
    targets: []
application:
    id: 1
    name: Customers
state: Succeeded
image: quay.io/konveyor/tackle2-addon-analyzer@sha256:fe3b25ecb831e66849aef89a24b11f935fe0a1230fcc44e1ab74ddf075c51870
pod: my-konveyor-operator/task-1-tlg6w
started: 2023-11-03T11:07:36.012089241Z
terminated: 2023-11-03T11:11:33.475575617Z
bucket:
    id: 11
    name: ""
activity:
    - Fetching application.
    - '[CMD] Running: /usr/bin/ssh-agent -a /tmp/agent.1'
    - '[CMD] succeeded.'
    - '[SSH] Agent started.'
    - '[GIT] Cloning: https://github.com/konveyor/tackle-testapp.git'
    - '[GIT] Using credentials (id=1) Git.'
    - '[FILE] Created /addon/.gitconfig.'
    - '[FILE] Created /addon/.git-credentials.'
    - '[CMD] Running: /usr/bin/git clone https://github.com/konveyor/tackle-testapp.git /addon/source/tackle-testapp'
    - '[CMD] succeeded.'
    - '[CMD] Running: /usr/bin/git checkout main'
    - '[CMD] succeeded.'
    - '[MVN] Using credentials (id=2) Maven.'
    - '[RULESET] fetching: id=1 (.discovery)'
    - '[RULESET] fetching: id=24 (openshift)'
    - '[RULESET] fetching (dep): id=20 (.technology-usage)'
    - '[RULESET] fetching: id=23 (openjdk17)'
    - '[RULESET] fetching: id=14 (openjdk11/openjdk8)'
    - '[RULESET] fetching: id=13 (openjdk7/oraclejdk7)'
    - '[RULESET] fetching: id=15 (openjdk17/openjdk11)'
    - '[RULESET] fetching: id=17 (os/windows)'
    - '[CMD] Running: /usr/bin/windup-shim convert --outputdir /addon/rules/converted /addon/rules'
    - '[CMD] succeeded.'
    - 'Settings: /addon/opt/settings.json'
    - '> - name: go'
    - '>   binaryPath: /usr/bin/generic-external-provider'
    - '>   initConfig:'
    - '>   - location: /addon/source/tackle-testapp'
    - '>     analysisMode: full'
    - '>     providerSpecificConfig:'
    - '>       dependencyProviderPath: /usr/bin/golang-dependency-provider'
    - '>       lspServerPath: /root/go/bin/gopls'
    - '>       name: go'
    - '> - name: java'
    - '>   binaryPath: /jdtls/bin/jdtls'
    - '>   initConfig:'
    - '>   - location: /addon/source/tackle-testapp'
    - '>     analysisMode: full'
    - '>     providerSpecificConfig:'
    - '>       bundles: /jdtls/java-analyzer-bundle/java-analyzer-bundle.core/target/java-analyzer-bundle.core-1.0.0-SNAPSHOT.jar'
    - '>       depOpenSourceLabelsFile: /usr/local/etc/maven.default.index'
    - '>       lspServerPath: /jdtls/bin/jdtls'
    - '>       mavenSettingsFile: /addon/opt/maven/settings.xml'
    - '> - name: builtin'
    - '>   initConfig:'
    - '>   - location: /addon/source/tackle-testapp'
    - '>     analysisMode: ""'
    - '> '
    - '[CMD] Running: /usr/bin/konveyor-analyzer --provider-settings /addon/opt/settings.json --output-file /addon/report.yaml --rules /addon/rules/files --rules /addon/rules/rulesets/1/rules --rules /addon/rules/rulesets/24/rules --rules /addon/rules/rulesets/20/rules --rules /addon/rules/rulesets/23/rules --rules /addon/rules/rulesets/14/rules --rules /addon/rules/rulesets/13/rules --rules /addon/rules/rulesets/15/rules --rules /addon/rules/rulesets/17/rules --rules /addon/rules/converted --label-selector (konveyor.io/target=cloud-readiness||konveyor.io/target=openjdk||konveyor.io/target=linux) --dep-label-selector !konveyor.io/dep-source=open-source'
    - '[CMD] succeeded.'
    - '[CMD] Running: /usr/bin/konveyor-analyzer-dep --provider-settings /addon/opt/settings.json --output-file /addon/deps.yaml'
    - '[CMD] succeeded.'
    - 'Analysis reported. duration: 8.739651ms'
    - '[TAG] Tagging Application 1.'
    - Facts updated.
    - Done.
```
